src/history/HistoryTestsUtils.cpp

```diff
@@ -89,6 +90,86 @@ S3HistoryConfigurator::configure(Config& mCfg, bool writable) const
     return mCfg;
 }
 
+BucketOutputIteratorForTesting::BucketOutputIteratorForTesting(
+    std::string tmpDir)
+    : BucketOutputIterator{tmpDir, false}
+{
+}
+
+std::pair<std::string, uint256>
+BucketOutputIteratorForTesting::writeTmpTestBucket(ArchiveFileState state)
+{
+    auto ledgerEntries =
+        LedgerTestUtils::generateValidLedgerEntries(NUM_ITEMS_PER_BUCKET);
+    auto bucketEntries = Bucket::convertToBucketEntry(ledgerEntries);
+
+    for (auto const& bucketEntry : bucketEntries)
+    {
+        put(bucketEntry);
+    }
+
+    // Finish writing and close the bucket file
+    assert(mBuf);
+    mOut.writeOne(*mBuf, mHasher.get(), &mBytesPut);
+    mObjectsPut++;
+    mBuf.reset();
+
+    if (state == ArchiveFileState::CORRUPTED_CONTENTS)
```

Comment:
There are 2 types of corruptions and bad data that we want to check:
* hash mismatch - which is what you ended up doing here but it would have been better to just change the hash by changing the last byte of the hash to a different value instead of changing the content
* good hash but the content is invalid (that will trip validation in different ways)
  * invalid because of duplicate content like what you did here ; really what we want is that the ledger entries are not sorted, so swapping two elements before writing to disk might be a better test than duplicate
  * invalid because of garbage (that doesn't represent a ledger entry and will trip marshaling code)